@nx/linter dependency is missing when using nx preset

### Please provide the environment you discovered this bug in.

v0.2.0-beta.26

### Which area/package is the issue in?

create-analog

### Description

Running
```sh
npx create-nx-workspace@latest --preset=@analogjs/platform
```
with the new release fails with the following error:

### Please provide the exception or error you saw

```true
create-nx-workspace@16.5.5
Ok to proceed? (y) y

 >  NX   Let's create a new workspace [https://nx.dev/getting-started/intro]

✔ Where would you like to create your workspace? · analog-nx-test
✔ Enable distributed caching to make your CI faster · No

 >  NX   Creating your v16.5.5 workspace.

   To make sure the command works reliably in all environments, and that the preset is applied correctly,
   Nx will run "npm install" several times. Please wait.

✔ Installing dependencies with npm
✔ Successfully created the workspace: analog-nx-test.
✔ What name would you like to use for your Analog app? · example
✔ Add TailwindCSS for styling? (Y/n) · true
✔ Add tRPC for typesafe client/server interaction? (y/N) · true

 >  NX   Cannot find module '@nx/linter'
```


### Other information

@nx/linter needs to be added during the preset process

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

--- a/packages/nx-plugin/src/generators/app/generator.spec.ts
+++ b/packages/nx-plugin/src/generators/app/generator.spec.ts
@@ -41,6 +41,9 @@ describe('nx-plugin generator', () => {
 
     expect(devDependencies['@nx/devkit']).toBe('~16.0.0');
     expect(devDependencies['@nx/angular']).toBe('~16.0.0');
+    // we just check for truthy because @nx/linter generator
+    // will install the correct version based on Nx version
+    expect(devDependencies['@nx/linter']).toBeTruthy();
     expect(devDependencies['@analogjs/platform']).toBe('0.1.0-beta.22');
     expect(devDependencies['@analogjs/vite-plugin-angular']).toBe(
       '0.2.0-alpha.29'
@@ -69,6 +72,9 @@ describe('nx-plugin generator', () => {
 
     expect(devDependencies['@nx/devkit']).toBe('^16.4.0');
     expect(devDependencies['@nx/angular']).toBe('^16.4.0');
+    // we just check for truthy because @nx/linter generator
+    // will install the correct version based on Nx version
+    expect(devDependencies['@nx/linter']).toBeTruthy();
     expect(devDependencies['@analogjs/platform']).toBe('^0.2.0-beta.26');
     expect(devDependencies['@analogjs/vite-plugin-angular']).toBe(
       '^0.2.0-beta.26'

--- a/packages/nx-plugin/src/generators/app/lib/initialize-analog-workspace.ts
+++ b/packages/nx-plugin/src/generators/app/lib/initialize-analog-workspace.ts
@@ -58,6 +58,7 @@ const initWithNxNamespace = async (
   try {
     ensurePackage('@nx/devkit', versions['@nx/devkit']);
     ensurePackage('@nx/angular', versions['@nx/angular']);
+    ensurePackage('@nx/linter', versions['@nx/linter']);
   } catch {
     // @nx/angular cannot be required so this fails but this will still allow executing the nx angular init later on
   }
@@ -67,6 +68,7 @@ const initWithNxNamespace = async (
     {
       '@nx/devkit': versions['@nx/devkit'],
       '@nx/angular': versions['@nx/angular'],
+      '@nx/linter': versions['@nx/linter'],
     }
   );
 
@@ -94,6 +96,7 @@ const initWithNrwlNamespace = async (
   try {
     ensurePackage('@nrwl/devkit', versions['@nrwl/devkit']);
     ensurePackage('@nrwl/angular', versions['@nrwl/angular']);
+    ensurePackage('@nrwl/linter', versions['@nrwl/linter']);
   } catch {
     // @nx/angular cannot be required so this fails but this will still allow executing the nx angular init later on
   }
@@ -103,6 +106,7 @@ const initWithNrwlNamespace = async (
     {
       '@nrwl/devkit': versions['@nrwl/devkit'],
       '@nrwl/angular': versions['@nrwl/angular'],
+      '@nrwl/linter': versions['@nrwl/linter'],
     }
   );
   await (

--- a/packages/nx-plugin/src/generators/app/versions/nx-dependencies.ts
+++ b/packages/nx-plugin/src/generators/app/versions/nx-dependencies.ts
@@ -1,14 +1,24 @@
 import { clean, lt } from 'semver';
 import { stripIndents } from '@nx/devkit';
-import { V16_X_NX_DEVKIT, V16_X_NX_ANGULAR } from './nx_16_X/versions';
+import {
+  V16_X_NX_DEVKIT,
+  V16_X_NX_ANGULAR,
+  V16_X_NX_LINTER,
+} from './nx_16_X/versions';
 import {
   V15_X_NRWL_DEVKIT,
   V15_X_NX_DEVKIT,
   V15_X_NRWL_ANGULAR,
   V15_X_NX_ANGULAR,
+  V15_X_NX_LINTER,
+  V15_X_NRWL_LINTER,
 } from './nx_15_X/versions';
 
-const nrwlDependencyKeys = ['@nrwl/devkit', '@nrwl/angular'] as const;
+const nrwlDependencyKeys = [
+  '@nrwl/devkit',
+  '@nrwl/angular',
+  '@nrwl/linter',
+] as const;
 export type NrwlDependency = (typeof nrwlDependencyKeys)[number];
 export const getNrwlDependencies = (
   nxVersion: string
@@ -27,6 +37,7 @@ export const getNrwlDependencies = (
     return {
       '@nrwl/angular': V15_X_NRWL_ANGULAR,
       '@nrwl/devkit': V15_X_NRWL_DEVKIT,
+      '@nrwl/linter': V15_X_NRWL_LINTER,
     };
   }
 
@@ -36,7 +47,7 @@ export const getNrwlDependencies = (
   );
 };
 
-const nxDependencyKeys = ['@nx/devkit', '@nx/angular'] as const;
+const nxDependencyKeys = ['@nx/devkit', '@nx/angular', '@nx/linter'] as const;
 export type NxDependency = (typeof nxDependencyKeys)[number];
 export const getNxDependencies = (
   nxVersion: string
@@ -55,6 +66,7 @@ export const getNxDependencies = (
     return {
       '@nx/angular': V15_X_NX_ANGULAR,
       '@nx/devkit': V15_X_NX_DEVKIT,
+      '@nx/linter': V15_X_NX_LINTER,
     };
   }
 
@@ -62,5 +74,6 @@ export const getNxDependencies = (
   return {
     '@nx/angular': V16_X_NX_ANGULAR,
     '@nx/devkit': V16_X_NX_DEVKIT,
+    '@nx/linter': V16_X_NX_LINTER,
   };
 };

--- a/packages/nx-plugin/src/generators/app/versions/nx_15_X/versions.ts
+++ b/packages/nx-plugin/src/generators/app/versions/nx_15_X/versions.ts
@@ -18,6 +18,8 @@ export const V15_X_PRISMJS = '^1.29.0';
 
 // devDependencies
 export const V15_X_ANALOG_JS_PLATFORM = '0.1.0-beta.22';
+export const V15_X_NRWL_LINTER = '~15.9.0';
+export const V15_X_NX_LINTER = '~16.0.0';
 export const V15_X_NRWL_VITE = '~15.9.0';
 export const V15_X_NX_VITE = '~16.0.0';
 export const V15_X_JSDOM = '^20.0.0';

--- a/packages/nx-plugin/src/generators/app/versions/nx_16_X/versions.ts
+++ b/packages/nx-plugin/src/generators/app/versions/nx_16_X/versions.ts
@@ -19,6 +19,7 @@ export const V16_X_PRISMJS = '^1.29.0';
 // devDependencies
 export const V16_X_ANALOG_JS_PLATFORM = '^0.2.0-beta.26';
 export const V16_X_NX_VITE = '^16.4.0';
+export const V16_X_NX_LINTER = '^16.4.0';
 export const V16_X_JSDOM = '^22.0.0';
 export const V16_X_VITE = '^4.3.9';
 export const V16_X_VITE_TSCONFIG_PATHS = '^4.2.0';

